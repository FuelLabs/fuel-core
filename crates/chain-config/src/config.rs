mod blob;
mod chain;
mod coin;
mod consensus;
mod contract;
mod message;
#[cfg(feature = "test-helpers")]
mod randomize;
#[cfg(feature = "std")]
mod snapshot_metadata;
mod state;
mod table_entry;

pub use blob::*;
pub use chain::*;
pub use coin::*;
pub use consensus::*;
pub use contract::*;
pub use message::*;
#[cfg(feature = "test-helpers")]
pub use randomize::*;
#[cfg(feature = "std")]
pub use snapshot_metadata::*;
pub use state::*;
pub use table_entry::*;
