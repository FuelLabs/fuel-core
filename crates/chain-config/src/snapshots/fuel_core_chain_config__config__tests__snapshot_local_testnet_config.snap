---
source: crates/chain-config/src/config.rs
expression: json
---
{
  "chain_name": "local_testnet",
  "block_gas_limit": 1000000000,
  "initial_state": {
    "coins": [
      {
        "owner": "0x6b63804cfbf9856e68e5b6e7aef238dc8311ec55bec04df774003a2c96e0418e",
        "amount": "0x0000000000989680",
        "asset_id": "0x0000000000000000000000000000000000000000000000000000000000000000"
      },
      {
        "owner": "0x54944e5b8189827e470e5a8bacfc6c3667397dc4e1eef7ef3519d16d6d6c6610",
        "amount": "0x0000000000989680",
        "asset_id": "0x0000000000000000000000000000000000000000000000000000000000000000"
      },
      {
        "owner": "0xe10f526b192593793b7a1559a391445faba82a1d669e3eb2dcd17f9c121b24b1",
        "amount": "0x0000000000989680",
        "asset_id": "0x0000000000000000000000000000000000000000000000000000000000000000"
      },
      {
        "owner": "0x577e424ee53a16e6a85291feabc8443862495f74ac39a706d2dd0b9fc16955eb",
        "amount": "0x0000000000989680",
        "asset_id": "0x0000000000000000000000000000000000000000000000000000000000000000"
      },
      {
        "owner": "0xc36be0e14d3eaf5d8d233e0f4a40b3b4e48427d25f84c460d2b03b242a38479e",
        "amount": "0x0000000000989680",
        "asset_id": "0x0000000000000000000000000000000000000000000000000000000000000000"
      }
    ]
  },
  "transaction_parameters": {
    "contract_max_size": 16777216,
    "max_inputs": 255,
    "max_outputs": 255,
    "max_witnesses": 255,
    "max_gas_per_tx": 100000000,
    "max_script_length": 1048576,
    "max_script_data_length": 1048576,
    "max_storage_slots": 255,
    "max_predicate_length": 1048576,
    "max_predicate_data_length": 1048576,
    "max_gas_per_predicate": 100000000,
    "gas_price_factor": 1000000000,
    "gas_per_byte": 4,
    "max_message_data_length": 1048576,
    "chain_id": 0
  },
  "gas_costs": {
    "add": 1,
    "addi": 1,
    "aloc": 1,
    "and": 1,
    "andi": 1,
    "bal": 22,
    "bhei": 1,
    "bhsh": 1,
    "burn": 126,
    "cb": 1,
    "cfei": 1,
    "cfsi": 1,
    "croo": 20,
    "div": 1,
    "divi": 1,
    "ecr": 1592,
    "eq": 1,
    "exp": 1,
    "expi": 1,
    "flag": 1,
    "gm": 1,
    "gt": 1,
    "gtf": 1,
    "ji": 1,
    "jmp": 1,
    "jne": 1,
    "jnei": 1,
    "jnzi": 1,
    "jmpf": 1,
    "jmpb": 1,
    "jnzf": 1,
    "jnzb": 1,
    "jnef": 1,
    "jneb": 1,
    "k256": 16,
    "lb": 1,
    "log": 43,
    "lt": 1,
    "lw": 1,
    "mcpi": 3,
    "mint": 127,
    "mlog": 1,
    "mod": 1,
    "modi": 1,
    "move": 1,
    "movi": 1,
    "mroo": 2,
    "mul": 1,
    "muli": 1,
    "mldv": 1,
    "noop": 1,
    "not": 1,
    "or": 1,
    "ori": 1,
    "ret_contract": 63,
    "rvrt_contract": 65,
    "s256": 4,
    "sb": 1,
    "scwq": 23,
    "sll": 1,
    "slli": 1,
    "srl": 1,
    "srli": 1,
    "srw": 21,
    "sub": 1,
    "subi": 1,
    "sw": 1,
    "sww": 67,
    "swwq": 68,
    "time": 1,
    "tr": 255,
    "tro": 196,
    "wdcm": 1,
    "wqcm": 1,
    "wdop": 1,
    "wqop": 1,
    "wdml": 1,
    "wqml": 2,
    "wddv": 2,
    "wqdv": 3,
    "wdmd": 4,
    "wqmd": 7,
    "wdam": 3,
    "wqam": 4,
    "wdmm": 4,
    "wqmm": 4,
    "xor": 1,
    "xori": 1,
    "call": {
      "base": 173,
      "dep_per_unit": 180
    },
    "ccp": {
      "base": 22,
      "dep_per_unit": 152
    },
    "csiz": {
      "base": 16,
      "dep_per_unit": 868
    },
    "ldc": {
      "base": 22,
      "dep_per_unit": 150
    },
    "logd": {
      "base": 48,
      "dep_per_unit": 18
    },
    "mcl": {
      "base": 1,
      "dep_per_unit": 2503
    },
    "mcli": {
      "base": 1,
      "dep_per_unit": 2559
    },
    "mcp": {
      "base": 1,
      "dep_per_unit": 1301
    },
    "meq": {
      "base": 1,
      "dep_per_unit": 1747
    },
    "retd_contract": {
      "base": 71,
      "dep_per_unit": 18
    },
    "smo": {
      "base": 207,
      "dep_per_unit": 18
    },
    "srwq": {
      "base": 44,
      "dep_per_unit": 5
    }
  },
  "consensus": {
    "PoA": {
      "signing_key": "22ec92c3105c942a6640bdc4e4907286ec4728e8cfc0d8ac59aad4d8e1ccaefb"
    }
  }
}
